docs/optlib.rst

```diff
@@ -181,269 +206,413 @@ Tips about writing option file
+you can safely use ``\n`` because that regex is not compiled with REG_NEWLINE.
+
+You should always test your regex patterns against test files with strings that
+do and do not match. Pay particular emphasis to when it should *not* match, and
+how *much* it matches when it should. A common error is forgetting that a
+POSIX.2 ERE engine is always greedy; the `*` and `+` quantifiers match
+as much as possible, before backtracking from the end of their match.
+
+For example this pattern::
+
+	foo.*bar
+
+Will match this **entire** string, not just the first part::
+
+	foobar, bar, and even more bar
+
+
+Regex option argument flags
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Even if more flags are added like ``x``, ``y``, ``z``,..., users
-may not utilize them well because it is difficult to memorize them. In
-addition, if many "option libraries" are contributed, we have to
-maintain them.
+Many regex-based options described in this document support additonal arguments
+in the form of long flags. Long flags are specified with surrounding ``{`` and
+``}``.
 
-For both users and developers the variety of short flags are just
-nightmares.
+The general format and placement is as follows::
 
-So universal-ctags now includes an API for defining long flags, which can be
-used as aliases for short flags. The long flags requires more typing
-but are more readable.
+	--regex-<LANG>=<PATTERN>/<NAME>/[<KIND>/]LONGFLAGS
 
-Here is the mapping between the standard short flag names and long flag names:
+Some examples:
 
-=========== ===========
-short flag  long flag
-=========== ===========
-b           basic
-e           extend
-i           icase
-=========== ===========
+.. code-block:: perl
 
-Long flags can be specified with surrounding ``{`` and ``}``.
-So the following ``--regex-<LANG>`` expression ::
+	--regex-Pod=/^=head1[ \t]+(.+)/\1/c/
+	--regex-Foo=/set=[^;]+/\1/v/{icase}
+	--regex-Man=/^\.TH[[:space:]]{1,}"([^"]{1,})".*/\1/t/{exclusive}{icase}{scope=push}
+	--regex-Gdbinit=/^#//{exclusive}
 
-   --m4-regex=/^m4_define\(\[([^]$\(]+).+$/\1/d,definition/x
+Note that the last example only has two ``/`` forward slashes following
+the regex pattern, as a shortened form when no kind-spec exists.
 
-is the same as ::
+The ``--mline-regex-<LANG>`` option also follows the above format. The
+experimental ``--_mtable-regex-<LANG>`` option follows a slightly
+modified version as well.
 
-   --m4-regex=/^m4_define\(\[([^]$\(]+).+$/\1/d,definition/{extend}
+The ``--langdef=<LANG>`` option also supports long flags, but not using
+forward-slash separators.
 
-The characters ``{`` and ``}`` may not be suitable for command line
-use, but long flags are mostly intended for option libraries.
+Regex control flags
+......................................................................
+
+.. Q: why even discuss the single-character version of the flags? Just
+	make everyone use the long form.
+
```

Comment:
I don't have strong reason.
The single character version (letter version) like `i` may be still used by users.
It is still useful if a flag is still appears frequently in a .ctags file.